src/appshell/view/dockwindow/dockpageview.cpp

```diff
@@ -188,7 +189,13 @@ void DockPageView::setDockOpen(const QString& dockName, bool open)
         return;
     }
 
-    if (!open) {
+    bool relevantPanel =
```

Comment:
You should not add local solutions to base classes/components. This class knows nothing about specific panels, like Palettes or Properties, and we shouldn't give it any such knowledge

Reply:
I'm new to the project and looking at picking up this issue. 

Instead of local name-based solutions like this, I imagine there should be some per-tab configuration property along the lines of "open makes hidden tab current". Does this pattern exist in MuseScore already? Or, is there a better existing way to data drive this?